testpar/t_file.c

```diff
@@ -1184,16 +1183,15 @@ test_fapl_preserve_hints(void)
     for (i = 0; i < nkeys_used; i++) {
 
         /* Memset the buffers to zero */
-        memset(key_used, 0, 20);
-        memset(value_used, 0, 20);
+        memset(key_used, 0, MPI_MAX_INFO_KEY + 1);
+        memset(value_used, 0, MPI_MAX_INFO_VAL + 1);
 
         /* Get the nth key */
         mpi_ret = MPI_Info_get_nthkey(info_used, i, key_used);
         VRFY((mpi_ret == MPI_SUCCESS), "MPI_Info_get_nthkey succeeded");
 
         if (!strcmp(key_used, key)) {
-
-            mpi_ret = MPI_Info_get(info_used, key_used, 20, value_used, &flag);
+            mpi_ret = MPI_Info_get(info_used, key_used, MPI_MAX_INFO_VAL, value_used, &flag);
```

Comment:
The standard says the value length specified here should be 1 less than the number of bytes allocated to make room for the NUL terminator